src/compiler/transformers/declarations.ts

```diff
@@ -366,6 +415,9 @@ export function transformDeclarations(context: TransformationContext) {
     }
 
     function trackReferencedAmbientModule(node: ModuleDeclaration, symbol: Symbol) {
+        // We forbid references in isolated declarations no need to report any errors on them
+        if (isolatedDeclarations) return;

```

Comment:
I wonder if some of these early returns should/could be assertions or debug failures. That might make it easier to find errors?

E.g. here it looks like `trackReferencedAmbientModule` is only called in `trackReferencedAmbientModuleFromImport` and in checker.ts. The checker tests if the function exists first. Could make it an assertion here, remove the function from the tracker and add either assertion or early return in `trackReferencedAmbientModuleFromImport`.

All of this could be wrong, though. I have very little context here.

Reply:
Yeah. Changed it to remove those for function in the symbol tracker.